src/main/java/org/mifos/processor/bulk/camel/routes/DeDuplicationRoute.java

```diff
@@ -0,0 +1,122 @@
+import static org.mifos.processor.bulk.zeebe.ZeebeVariables.DE_DUPLICATION_FAILED;
+import static org.mifos.processor.bulk.zeebe.ZeebeVariables.DUPLICATE_TRANSACTION_COUNT;
+import static org.mifos.processor.bulk.zeebe.ZeebeVariables.DUPLICATE_TRANSACTION_FILE;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import org.mifos.processor.bulk.schema.Transaction;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DeDuplicationRoute extends BaseRouteBuilder {
+
+    @Override
+    public void configure() throws Exception {
+        from(RouteId.DE_DUPLICATION.getValue()).id(RouteId.DE_DUPLICATION.getValue())
+                .log("Started route " + RouteId.DE_DUPLICATION.getValue()).to("direct:download-file").to("direct:get-transaction-array")
+                .process(exchange -> {
+                    List<Transaction> transactionList = exchange.getProperty(TRANSACTION_LIST, List.class);
+
+                    if (Objects.isNull(transactionList) || transactionList.isEmpty()) {
+                        exchange.setProperty(DE_DUPLICATION_FAILED, false);
+                        exchange.setProperty(DUPLICATE_TRANSACTION_COUNT, 0);
+                    }
+
+                    int duplicateTxnCount = 0;
+                    List<Transaction> duplicateTransactionList = new ArrayList<>(); // contains the duplicate
+                                                                                    // transaction
+                    List<Transaction> originalTransactionList = new ArrayList<>(); // contains the original txn after
+                                                                                   // removing duplicate
+                    Set<String> set = new HashSet<>();
+
+                    for (Transaction transaction : transactionList) {
+                        String payeeDetail = fetchPayeeDetail(transaction);
+                        if (set.contains(payeeDetail)) {
+                            transaction.setNote("Duplicate transaction.");
+                            duplicateTransactionList.add(transaction);
+                            duplicateTxnCount++;
+                        } else {
+                            set.add(payeeDetail);
+                            originalTransactionList.add(transaction);
+                        }
+                    }
+
+                    log.info("Duplicate txn: {} and count: {}", duplicateTransactionList, duplicateTxnCount);
+
+                    exchange.setProperty(DUPLICATE_TRANSACTION_COUNT, duplicateTxnCount);
+                    exchange.setProperty(DUPLICATE_TRANSACTION_LIST, duplicateTransactionList);
+                    exchange.setProperty(ORIGINAL_TRANSACTION_LIST, originalTransactionList);
+
+                    // todo update the existing file , create a new duplicate txn file and upload in s3
```

Comment:
finish or remove the todo task

Reply:
I missed the comment, the task is already done.